CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+## 2.29.0-rc.0 / 2021-07-29
+
+* [CHANGE] Remove experimental tag on `--storage.tsdb.allow-overlapping-blocks` flag. #9117
+* [CHANGE] Remove experimental tag on `--storage.tsdb.retention.size` flag. #8728
+* [FEATURE] Add Kuma service discovery. #8844
+* [FEATURE] Add `__meta_gce_interface_ipv4_<name>` meta label to GCE discovery. #8978
+* [FEATURE] Add `__meta_ec2_availability_zone_id` meta label to EC2 discovery. #8978
+* [FEATURE] Add `__meta_hetzner_hcloud_labelpresent_<labelname>` meta label to Hetzner discovery. #9028
+* [FEATURE] Allow configuring exemplar storage via file and make it reloadable. #8974
+* [FEATURE] promtool: Add feature flags flag `--enable-feature`. #8958
+* [FEATURE] promtool: Add service discovery configuration validation. #8950
+* [ENHANCEMENT] Allow filtering with matchers in label-names API. #9083
```

Comment:
This on the other hand is something I'd be tempted to call a `FEATURE` due to its potentially big significance to API users, but @roidelapluie would maybe disagree with me again :)

Reply:
This is not a change for users, just now we filter at TSDB layer instead of API layer, if I remember correctly.

Reply:
I see, I thought it was an actual new feature, not just an improvement of the implementation. Although the pull request (https://github.com/prometheus/prometheus/pull/9083) mentions big resource savings (which we usually do surface in changelogs), but probably the label names endpoint is not significant enough for this.